shotover/src/transforms/query_counter.rs

```diff
@@ -26,11 +29,36 @@ pub struct QueryCounterConfig {
 
 impl QueryCounter {
     pub fn new(counter_name: String) -> Self {
-        counter!("shotover_query_count", "name" => counter_name.clone());
+        let counter_name_ref: &'static str = counter_name.leak();
+        let mut counters = HashMap::new();
+
+        let query_counter_key = format!("shotover_query_count-name:{}", counter_name_ref);
+        let query_counter_key_ref: &'static str = query_counter_key.leak();
+        let query_counter = counter!("shotover_query_count", "name" => counter_name_ref);
+        counters.insert(query_counter_key_ref, query_counter);
 
         QueryCounter {
             // Leaking here is fine since the builder is created only once during shotover startup.
-            counter_name: counter_name.leak(),
+            counter_name: counter_name_ref,
+            counters,
+        }
+    }
+
+    fn increment_counter(&mut self, query: &'static str, query_type: &'static str) {
+        let query_counter_key = format!(
```

Comment:
We should be able to use `(&'static str, &'static str)` as a key instead of constructing a new string

Reply:
In that case, how can we differentiate between the counters with `query=unknown`?

Reply:
`query=unknown` is using the query field which is the field I am proposing we keep, so it should work fine.

Reply:
We still want to include the "type" field when we create the counter.
However we do not need to worry about different type values colliding in the hashmap since a single transform can only process a single protocol. e.g. it will process only redis message or only kafka messages.

The only exception to this is `Frame::Dummy` which can be sent by any protocol but does not record to a counter since its a shotover implementation detail.